log.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2015 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package trace implements utility functions for capturing logs
+package trace
+
+import (
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"runtime"
+)
+
+const (
+	// LineField is a field with code line added to structured traces
+	LineField = "line"
+	// FileField is a field with code file added to structured traces
+	FileField = "file"
+)
+
+// TextFormatter is logrus-compatible formatter, adding
+// file and line to every logged entry
+type TextFormatter struct {
+	log.TextFormatter
+}
+
+// Format implements logrus.Formatter interface and adds file and line
```

Comment:
`... and adds file and line details` (not just `file and line`)

If you wanted to enforce that the type implements a specific interface, then a construct like:
```go
var _ log.Formatter = (*TextFormatter)(nil)
```
is an explicit statement of that. The same goes for `JSONFormatter`.